pkg/networkservice/mechanisms/vxlan/option.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2020-2021 Cisco and/or its affiliates.
+//
+// Copyright (c) 2021 Nordix Foundation.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vxlan
+
+import (
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/mechanisms/vxlan/vni"
+)
+
+// Option is an option pattern for vxlan server/client
+type Option func(o *vxlanOpions)
+
+// WithVniOptions sets vni options
+func WithVniOptions(vniOptions ...vni.Option) Option {
```

Comment:
Could we please just make this a WithVniPort that takes an int.  Passing layers of Options cross binds us at the API level in an unpleasant way.  Its easy enough to translate the options in the NewClient/NewServer functions.